src/wallet/wallet.cpp

```diff
@@ -1506,6 +1507,29 @@ int CWallet::ScanForWalletTransactions(CBlockIndex* pindexStart, bool fUpdate)
     return ret;
 }
 
+void CWallet::ScanForWalletUTXOs()
+{
+    for (std::unique_ptr<CCoinsViewCursor> pcursor(pcoinsTip->Cursor());
```

Comment:
Would there be a better way to keep the wallet separated from the UTXO set?
During wallet creation (`init.cpp`) it could pass-in a lambda that would take care about finding the corresponding `CCoins`.